rider-fsharp/protocol/src/kotlin/model/RdFSharpTypeProvidersLoaderModel.kt

```diff
@@ -0,0 +1,291 @@
+package model
+
+import com.jetbrains.rider.model.nova.ide.SolutionModel
+import com.jetbrains.rd.generator.nova.*
+import com.jetbrains.rd.generator.nova.PredefinedType.*
+import com.jetbrains.rd.generator.nova.csharp.CSharp50Generator
+import java.io.File
+
+@Suppress("unused")
+object RdFSharpTypeProvidersLoaderModel : Root(
+        CSharp50Generator(FlowTransform.AsIs, "JetBrains.Rider.FSharp.TypeProvidersProtocol.Server", File("C:\\Programming\\fsharp-support\\ReSharper.FSharp\\src\\FSharp.TypeProvidersProtocol\\src\\Server")),
+        CSharp50Generator(FlowTransform.Reversed, "JetBrains.Rider.FSharp.TypeProvidersProtocol.Client", File("C:\\Programming\\fsharp-support\\ReSharper.FSharp\\src\\FSharp.TypeProvidersProtocol\\src\\Client"))
+) {
+    lateinit var RdProvidedType: Class.Concrete
+    lateinit var RdProvidedPropertyInfo: Class.Concrete
+    lateinit var RdProvidedMethodInfo: Class.Concrete
+    lateinit var RdProvidedParameterInfo: Class.Concrete
+
+    private val RdProvidedMemberProcessModel = baseclass {
+        call("DeclaringType", int, RdProvidedType.nullable)
+    }
+
+    private val RdProvidedMethodBaseProcessModel = baseclass extends RdProvidedMemberProcessModel {
+        call("GetParameters", int, array(RdProvidedParameterInfo))
+        call("GetGenericArguments", int, array(RdProvidedType))
+        //call("GetStaticParametersForMethod", RdTypeProvider, array(RdProvidedParameterInfo))
+    }
+
+    private val RdProvidedMemberInfo = baseclass {
+        field("Name", string)
+        field("EntityId", int)
+    }
+
+    private val RdProvidedMethodBase = baseclass extends RdProvidedMemberInfo {
+        field("IsGenericMethod", bool)
+        field("IsStatic", bool)
+        field("IsFamily", bool)
+        field("IsFamilyAndAssembly", bool)
+        field("IsFamilyOrAssembly", bool)
+        field("IsVirtual", bool)
+        field("IsFinal", bool)
+        field("IsPublic", bool)
+        field("IsAbstract", bool)
+        field("IsHideBySig", bool)
+        field("IsConstructor", bool)
+    }
+
+    private val RdResolutionEnvironment = structdef {
+        field("resolutionFolder", string)
+        field("outputFile", string.nullable) //string option
+        field("showResolutionMessages", bool)
+        field("referencedAssemblies", array(string))
+        field("temporaryFolder", string)
+    }
+
+    private val RdProvidedNamespaceProcessModel = aggregatedef("RdProvidedNamespaceProcessModel") {
+        call("GetNestedNamespaces", int, array(RdProvidedNamespace))
+        call("GetTypes", int, array(int))
+        call("ResolveTypeName", structdef("ResolveTypeNameArgs") {
+            field("Id", int)
+            field("TypeFullName", string)
+        }, int)
+    }
+
+    private val RdProvidedNamespace = classdef {
+        field("NamespaceName", string)
+        field("EntityId", int)

```

Comment:
Could it be just `Id` like in some other models? And, well, let's use a single word across these types. :)